serial_protocol.md

```diff
@@ -0,0 +1,72 @@
+#Protocol:
+During the first use, or when the chiplet changes environments a “NETWORK” call is expected to initialize the wifi parameters.
+
+Every time a serial connection is established we will expect a “INIT” call after which any subsequent calls can be made, until the connection is closed. 
+
+In the following examples we use $ to represent variables. For example $Host represents your firebase host.
+
+##Response Format Byte
+All responses will be prefixed with one of the following bytes signifying the response type.
+```
+  + If response is ok and a raw value
+  * If response is ok and a raw value prefixed by count of bytes in response. 
```

Comment:
actually as I think about handling multiline strings will be a pain anyway. I think we should just pass them back to the client and allow them to un-escape the newlines. 
Since we don't know how many new lines there are we cannot just use the http body size as the size of our payload since everytime we translate the escaped line to a \n we will reduce the size. Technically we can just pad the output, since we would always be the stated size or less it shouldn't cause a problem but it seems extra work for something that isn't actually hard for the client to do. 
Also the closer we keep our wireformat to the response from the REST call the better I think. That way they can just use curl for debugging to see what the value should look like.

Alright I am sold, lets just determine a threshold to autopromote a string to a bulk string.